src/clr-angular/forms/combobox/option-items.directive.spec.ts

```diff
@@ -0,0 +1,206 @@
+ */
+import { Component, ViewChild, Renderer2 } from '@angular/core';
+import { TestBed } from '@angular/core/testing';
+import { ClrOptionItems } from './option-items.directive';
+import { ClrComboboxModule } from './combobox.module';
+import { OptionSelectionService } from './providers/option-selection.service';
+import { ClrPopoverPositionService } from '../../utils/popover/providers/popover-position.service';
+import { ClrPopoverEventsService } from '../../utils/popover/providers/popover-events.service';
+import { ClrPopoverToggleService } from '../../utils/popover/providers/popover-toggle.service';
+
+@Component({
+  template: `
+    <ul>
+      <li *clrOptionItems="let n of numbers; trackBy: trackBy">{{ n }}</li>
+    </ul>
+  `,
+})
+class FullTest {
+  @ViewChild(ClrOptionItems) optionItems: ClrOptionItems<number>;
+  numbers = [1, 2, 3, 4, 5];
+  trackBy: (index: number, item: number) => any;
+}
+
+@Component({
+  template: `
+    <ul>
+      <li *clrOptionItems="let n of numbers; trackBy: trackBy">{{ n }}</li>
+    </ul>
+  `,
+})
+class TrackByIndexTest {
+  @ViewChild(ClrOptionItems) optionItems: ClrOptionItems<number>;
+  numbers = [1, 2, 3, 4, 5];
+  trackBy = (index: number) => index;
+}
+
+@Component({
+  template: `
+    <ul>
+      <li *clrOptionItems="let n of numbers; field:'a'">{{ n.a }}</li>
+    </ul>
+  `,
+})
+class ObjectDataTest {
+  @ViewChild(ClrOptionItems) optionItems: ClrOptionItems<number>;
+  numbers = [{ a: 1 }, { a: 2 }, { a: 3 }];
+  trackBy = (index: number) => index;
+}
+
+const OPTION_ITEM_PROVIDERS = [
+  OptionSelectionService,
+  ClrPopoverToggleService,
+  ClrPopoverPositionService,
+  ClrPopoverEventsService,
+  Renderer2,
+];
+
+export default function(): void {
+  describe('ClrOptionItems directive', function() {
+    describe('correctly initializes', () => {
+      beforeEach(function() {
+        TestBed.configureTestingModule({
+          imports: [ClrComboboxModule],
+          declarations: [FullTest],
+          providers: OPTION_ITEM_PROVIDERS,
+        });
+        this.fixture = TestBed.createComponent(FullTest);
+        this.fixture.detectChanges();
+        this.testComponent = this.fixture.componentInstance;
+        this.clarityDirective = this.testComponent.optionItems;
+      });
+
+      afterEach(function() {
+        this.fixture.destroy();
+      });
+
+      it('can handle changes', function() {
+        const initialContent = this.fixture.elementRef.nativeElement.textContent;
+        expect(initialContent.trim()).toEqual('12345');
+        this.testComponent.numbers.push(6);
+        this.fixture.detectChanges();
+        const updatedContent = this.fixture.elementRef.nativeElement.textContent;
+
+        /* This took me quite some time to research, so it needs a detailed explanation.
+           The data not updating immediately does not mean that the new value will not be added to the combobox.
```

Comment:
I suspect that this happens because the updateItems method isn't telling the iterableProxt to run ngDoCheck.

Reply:
Moving the call of ngDoCheck to the updateItems method had no effect on this test.

Reply:
We talked and agreed that this issue is not related to the way that the directive works but that we're not monitoring for internal changes in the data array. This, so far, is acceptable, as it gets updated on popover reopening and popovers staying consistent while open is actually desired, even if there is a chance to get this data updated.